src/NuGet.Clients/NuGet.PackageManagement.UI/Utility/UIUtility.cs

```diff
@@ -47,33 +47,51 @@ private static bool IsHttpUrl(Uri uri)
             return (uri.Scheme == Uri.UriSchemeHttp || uri.Scheme == Uri.UriSchemeHttps);
         }
 
-        private static readonly string[] _scalingFactor = new string[] {
-            String.Empty,
-            "K", // kilo
-            "M", // mega, million
-            "G", // giga, billion
-            "T"  // tera, trillion
-        };
-
         // Convert numbers into strings like "1.2K", "33.4M" etc.
         // Precondition: number > 0.
         public static string NumberToString(long number, IFormatProvider culture)
         {
-            double v = (double)number;
-            int exp = 0;
+            double RoundDown(double num, long precision)
+            {
+                double val = num / precision;
+                return val > 999 ? 999 : val;
+            }
+
+            string GetFormatted(double num, string precisionString)
+            {
+                return string.Format(
+                   culture,
+                   "{0:G3}{1}",
```

Comment:
Wouldn't it be better to include the `{0:G3}` in the localized string?  Maybe some cultures prefer to have a space between the number and the suffix, while others don't. For example `123 M` vs `123M`

Reply:
@zivkan 
I moved into localized string, please review again.